### PR TITLE
Exclude net-sftp lib

### PR DESCRIPTION
**Pull request by askonev**
Additionally, the use of ssh libraries below version 7.0.0 is strictly blocked because openssl 3.0 was previously used.

Maybe on Ubuntu below 22.04 where openssl 3.0 is not used everything will break. We'll have to check it out.

### CODE REVIEW

File: lib/data/static_data.rb [hunk 1]
```diff
@@ -20,6 +20,8 @@ class StaticData
     { dir: '.documentserver', file: 'documentserver_jwt' }
   ].freeze
 
+  TMP = "#{Dir.pwd}/tmp"
```
Comment (ShockwaveNN) [inline lib/data/static_data.rb:23]: Maybe use system tmp directory? Via  `Dir.mktmpdir` or something like this?

**Review by ShockwaveNN — CHANGES_REQUESTED** [review]

File: lib/helpers/remote_configuration.rb [hunk 2]
```diff
@@ -27,21 +29,40 @@ def f_manager
...
-    ssh.upload!(session,
-                StaticData::ENV,
-                env = f_manager.overwrite(env, /latest/, f_manager.wrap_in_double_quotes(@version)))
-    ssh.upload!(session,
-                StaticData::ENV,
-                f_manager.overwrite(env, /''/, f_manager.wrap_in_double_quotes(@spec_name)))
+  # @param [Object] tmpdir
+  # @return [Integer]
+  #
+  def overwrite_dockerfile(tmpdir)
```
Comment (ShockwaveNN) [inline lib/helpers/remote_configuration.rb:35]: I think sending full path to `Dockerfile` in this method will be more precise, than sending dir and *hoping* that there is a Dockerfile inside

Same for `dot_env`

**Review by ShockwaveNN — CHANGES_REQUESTED** [review]

**Review by ShockwaveNN — APPROVED** [review]

**Comment by askonev** [comment]
Apparently the DO has some sort of tricky ssh ban policy. If you reduce the wait after sftp request to less than 5 seconds, then the ban is on the fourth "fast" request. But I could not wait for the unbanning. So hardcode at sleep 5 is a normal solution. But sometimes it bans even for such time interval.

File: lib/ssh/ssh_wrapper.rb [hunk 3]
```diff
@@ -33,22 +32,32 @@ def exec_with_logs!(session, command)
...
   # @param [Object] remote_path
   # @param [Object] user
   # @param [Object] ip
-  # @return [Object] recursion
+  # @return [Integer]
   def sftp_put(host_path, remote_path, user, ip)
-    system("echo \"put #{host_path} #{remote_path}\" | sftp #{user}@#{ip}")
-    sleep 5 # Timeout between commands to not be banned by sftp
-    sftp_put(host_path, remote_path, user, ip) unless $CHILD_STATUS.success?
+    5.times do
```
Comment (ShockwaveNN) [inline lib/ssh/ssh_wrapper.rb:53]: Maybe extract similar code in `sftp_get` in `sftp_put` in new method and call it inside sftp_put something like
```
def sftp_put
  sftp_command("echo \"put #{host_path} #{remote_path}\" | sftp #{user}@#{ip}")
end
 ```

**Review by ShockwaveNN — CHANGES_REQUESTED** [review]

**Review by ShockwaveNN — APPROVED** [review]

**Review by l8556 — APPROVED** [review]

**Pull request merged**